packages/mui-joy/src/Link/Link.tsx

```diff
@@ -216,6 +262,10 @@ Link.propTypes /* remove-proptypes */ = {
    * @ignore
    */
   onFocus: PropTypes.func,
+  /**
+   * Element placed before the children.
+   */
+  startIcon: PropTypes.node,
```

Comment:
Question: Are you going to rename start/end Icon props in Link, Typography, etc to start/end Decorator as well?

Reply:
Honestly, I still doubt it. The word decorator is quite long to type. For Button, I don't see the use case where decorator can be other things than icons. However, Typography & Link might have a badge at the end.

@danilo-leal Should we always use `startDecorator` & `endDecorator` for all components, or should it depends on the type of component? @hbjORbj What do you think?

Reply:
What is the reason that we want to use `start/endDecorator` for `Button` instead of `start/endIcon`?

Reply:
> For Button, I don't see the use case where decorator can be other things than icons.

Does this mean that there is no reason to rename `start/endIcon` to `start/endDecorator`?

Reply:
Let's wait for @danilo-leal comment. If I have to choose, I would change to `start/endDecorator` for consistency.

```js
<Button startDecorator={...} endDecorator={...}>
<Typography startDecorator={...} endDecorator={...}>
<Link startDecorator={...} endDecorator={...}>
<Input startDecorator={...} endDecorator={...}>
```

Reply:
I'd say that whatever way we choose, we should keep consistency! _Decorator_ is broader than _Icon_. Is there a case where we'd want something other than an icon in these components?